File: 김은지/완주하지-못한-선수.js
function solution(participant, completion) {
  const map = new Map();

  for (const person of participant) {
    map.set(person, (map.get(person) || 0) + 1);
  }

  for (const person of completion) {
    map.set(person, map.get(person) - 1);
    if (map.get(person) === 0) map.delete(person);
  }

  return map.keys().next().value;
}

File: 김현경/완주하지_못한_선수.py
def solution(participant, completion):
    name_counts = {}
    
    #참가자 카운팅
    for p in participant:
        if p in name_counts:
            name_counts[p] += 1
        else:
            name_counts[p] = 1

    #완주자 카운트 감소        
    for c in completion:
        if c in name_counts:
            name_counts[c] -= 1

    # 완주자 카운트가 1인 사람이 완주하지 못한 사람    
    for key, value in name_counts.items():
        if value == 1:
            return key
